portal-impl/src/com/liferay/portal/webserver/WebServerServlet.java

```diff
@@ -1586,6 +1588,35 @@ private void _checkFileEntry(
 		}
 	}
 
+	private void _checkInstanceGroupId(
+			long groupId, HttpServletRequest httpServletRequest)
+		throws Exception {
+
+		Group group = GroupLocalServiceUtil.fetchGroup(groupId);
+
+		if (group == null) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(
+					StringBundler.concat("Group not found: ", groupId, "."));
+			}
+
+			throw new NoSuchFileEntryException("No file entry found");
+		}
+
+		long companyId = PortalUtil.getCompanyId(httpServletRequest);
+
+		if (!(group.getCompanyId() == companyId)) {
```

Comment:
Why not use `group.getCompanyId() != companyId` here?

Reply:
Completely agree, I must have changed it trying other conditions and I did not change it again

Reply:
Changed